File: app/src/main/java/com/blues/database/course/CourseDAO.java
package com.blues.database.course;


import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface CourseDAO {

    @Insert
    void insert(Course course);

    @Update
    void update(Course course);


    @Delete
    void delete(Course course);

    @Query("SELECT * FROM course")
    List<Course> getAll();

    @Query("SELECT * FROM course WHERE name LIKE :courseName")
    List<Course> getCourseByName(String courseName);
}

File: app/src/main/java/com/blues/course/database/CourseDAO.kt
package com.blues.course.database

import androidx.room.*
import com.blues.course.model.Course

@Dao
interface CourseDAO {
    @Insert
    fun insert(course: Course?)

    @Update
    fun update(course: Course?)

    @Delete
    fun delete(course: Course?)

    @get:Query("SELECT * FROM course")
    val fetchAll: List<Course>?

    @Query("SELECT * FROM course WHERE name LIKE :courseName")
    fun getCourseByName(courseName: String?): List<Course?>?
}